### PR TITLE
ci: Use idf-build-apps from example builds

### PR DESCRIPTION
**Pull request by tore-espressif**
# Change description

This new GitHub action will build all examples in this repository.

TODO:
- [x] CBOR examples have their own GH action; merge it with this one
- [x] Add support for example tests
- [x] Wait until the 'exclude feature' is fixed in idf-build-apps _EDIT:_ Fixed in idf-build-apps v1.0.0

### CODE REVIEW

**Comment by tore-espressif** [comment]
cc @kumekay @hfudev

File: .github/workflows/build_and_run_examples.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+          submodules: 'true'
+      - name: Build Examples
+        shell: bash
+        run: |
+          . ${IDF_PATH}/export.sh
+          pip install idf-component-manager idf-build-apps --upgrade
+          export PEDANTIC_FLAGS="-DIDF_CI_BUILD -Werror -Werror=deprecated-declarations -Werror=unused-variable -Werror=unused-but-set-variable -Werror=unused-function"
+          export EXTRA_CFLAGS="${PEDANTIC_FLAGS} -Wstrict-prototypes"
+          export EXTRA_CXXFLAGS="${PEDANTIC_FLAGS}"
+          idf-build-apps find
```
Comment (hfudev) [inline .github/workflows/build_and_run_examples.yml:31]: you don't need find to build it.

**Comment by hfudev** [comment]
LGTM!

File: .github/workflows/build_and_run_examples.yml (hunk 1 above)
Comment (tore-espressif) [reply .github/workflows/build_and_run_examples.yml:31]: Thank you for the comment. In this case I just want to have a summary of selected apps in the log, before we start the build process

**Review by kumekay — APPROVED** [review]
@tore-espressif Thank you, LGTM

**Review by mahavirj — APPROVED** [review]

**Review by kumekay — APPROVED** [review]
LGTM, thank you @tore-espressif

**Pull request merged**